src/data-workspace/data-entry-cell/data-entry-cell.js

```diff
@@ -0,0 +1,144 @@
+    const { input, meta } = useField(fieldName, { validate })
+
+    const [lastSyncedValue, setLastSyncedValue] = useState(meta.initial)
+    const { focusNext, focusPrev } = useFieldNavigation(fieldName)
+
+    const [mutate, { called, loading, error }] =
+        useDataMutation(DATA_VALUE_MUTATION)
+    const [dataEntryContext] = useContextSelection()
+    const { metadata } = useMetadata()
+
+    const syncData = () => {
+        const {
+            dataSetId,
+            orgUnitId,
+            periodId,
+            attributeOptionComboSelection,
+        } = dataEntryContext
+
+        const attributeComboId = getDataSetById(metadata, dataSetId)
+            .categoryCombo.id
+        const isDefaultAttributeCombo =
+            metadata.categoryCombos[attributeComboId].isDefault
+
+        const mutationVars = {
+            de: de.id,
+            co: coc.id,
+            ds: dataSetId,
+            ou: orgUnitId,
+            pe: periodId,
+            value: input.value,
+        }
+        // Add attribute params to mutation if relevant
+        if (!isDefaultAttributeCombo) {
+            // convert ['<catId>:<catOptId>', ...] into 'catOptId1;catOptId2':
+            const attributeOptionIdList = attributeOptionComboSelection
+                .map((aoc) => aoc.split(':')[1])
+                .join(';')
+            mutationVars.cc = attributeComboId
+            mutationVars.cp = attributeOptionIdList
+        }
+
+        mutate(mutationVars)
+        setLastSyncedValue(input.value)
+    }
+
+    const onBlur = (event) => {
+        // If this value has changed, sync it to server if valid
+        if (meta.dirty && input.value !== lastSyncedValue && meta.valid) {
+            syncData()
+        }
+        // Also invoke FinalForm's `onBlur`
+        input.onBlur(event)
+    }
+
+    const onKeyDown = (event) => {
+        const { key, shiftKey } = event
+        if (key === 'Enter' && shiftKey) {
+            // todo: open data item details
+        } else if (key === 'ArrowDown' || key === 'Enter') {
+            event.preventDefault()
+            focusNext()
+        } else if (key === 'ArrowUp') {
+            event.preventDefault()
+            focusPrev()
+        }
+        // tab and shift-tab on their own work as expected
+    }
+
+    // todo: get data details (via getDataValue?)
+    // todo: on focus, set 'active cell' in context
+    // todo: tooltip for invalid cells
+    // todo: validate with `de.valueType` (started)
+    // todo: implement other input types for different value types
+    // todo: implement read-only cells
+
+    const synced = meta.valid && called && !loading && !error
+    const inputStateClassName = meta.invalid
+        ? styles.inputInvalid
+        : synced
+        ? styles.inputSynced
+        : null
+
+    return (
+        <td className={styles.dataEntryCell}>
+            <div className={styles.cellInnerWrapper}>
+                <input
```

Comment:
I'm pretty sure you need to use the react-finalforms `Field`-component for all the FinalForm stuff to work...
I'm sure this is the reason why `form.focus(field)` in `use-field-navigation` does not work - you're not connecting the field to the form in anyway as far as I can tell.

Reply:
Nvm, Im stupid - didn't see the `useField`-hook.